Automattic/sensei
# WPML: Shared student progress and quiz submission

Resolves #2788 

Unfortunately, questions are not translated automatically. (You can see these strings in the WPML translation editor, but they don't appear in the translated lesson.)
For now, the user has to translate them manually in lessons: open the translated lesson in the WordPress editor and change texts for questions.

## Proposed Changes
* Share course/lesson/quiz progress across different translations of the same course/lesson/quiz.
* Share quiz submission.

## Testing Instructions

0. Install and activate WPML, add the second language.
1. Create three courses with lessons and quizzes.
2. Create translations for each of them.
3. Go to the first translated course on the frontend, complete it. There should be no issues.
4. Go to the backend, check in Students and Grading there is the student's progress attached to the original course/lesson/quiz.
5. Go to Sensei > Settings > Experimental Features and enable HPPS. Enable sync, use "classic" storage.
6. Go to the second translated course on the frontend, complete it. No issues expected.
7. Go to the backend, check in Students and Grading there is the student's progress attached to the original course/lesson/quiz.
8. Go to Sensei > Settings > Experimental Features and choose custom tables as a storage for progress.
9. Go to the third translated course on the frontend, complete it. No issues expected.
7. Go to the backend, check in Students and Grading there is the student's progress attached to the original course/lesson/quiz.

## New/Updated Hooks


* `sensei_block_take_course_course_id` - Filters the course ID for the take course block.
* `sensei_course_is_user_enrolled_course_id` - Filters the course ID for the `Sensei_Course::is_user_enrolled` method.
* `sensei_course_start_course_id` - Filter the course ID for the course being started.
* `sensei_utils_check_for_activity_before_get_comments` - This action runs before getting the comments for the given request.
* `sensei_utils_check_for_activity_after_get_comments` - This action runs after getting the comments for the given request.
* `sensei_utils_user_completed_lesson_lesson_id` - Filter lesson ID for `Sensei_Utils::user_completed_lesson` method.
* `sensei_course_manual_enrolment_enroll_learner_course_id` - Filter the course ID for manual enrolment when the student is being added to the course.
* `sensei_course_manual_enrolment_withdraw_learner_course_id` - Filter the course ID for manual enrolment when the student is being removed from the course.
* `sensei_quiz_answer_create_submission_id` - Filters the submission ID when quiz answer is created.
* `sensei_quiz_answer_create_question_id` - Filters the question ID when quiz answer is created.
* `sensei_quiz_answer_get_all_submission_id` - Filters the submission ID when getting all quiz answers.
* `sensei_quiz_answer_delete_all_submission_id` - Filters the submission ID when deleting all quiz answers.
* `sensei_quiz_grade_create_submission_id` - Filters the submission ID when quiz grade is created.
* `sensei_quiz_grade_create_question_id` - Filters the question ID when quiz grade is created.
* `sensei_quiz_grade_get_all_submission_id` - Filter the submission ID when getting all quiz grades.
* `sensei_quiz_grade_save_many_submission_id` - Filters the submission ID when saving many quiz grades.
* `sensei_quiz_grade_save_many_question_id` - Filters the question ID when saving many quiz grades.
* `sensei_quiz_grade_delete_all_submission_id` - Filters the submission ID when deleting all quiz grades.
* `sensei_quiz_submission_create_quiz_id` - Filters the quiz ID when quiz submission is created.
* `sensei_quiz_submission_get_or_create_quiz_id` - Filters the quiz ID when quiz submission is created.
* `sensei_quiz_submission_get_quiz_id` - Filters the quiz ID when quiz submission is retrieved.
* `sensei_quiz_submission_get_question_ids_submission_id` - Filters the quiz submission ID when getting the question IDs.
* `sensei_course_progress_create_course_id` - Filter the course ID for a created course progress.
* `sensei_course_progress_get_course_id` - Filter the course ID for a course progress we want to get.
* `sensei_course_progress_has_course_id` - Filter the course ID for a course progress we want to check.
* `sensei_course_progress_delete_for_course_course_id` - Filter the course ID for a course progress we want to delete.
* `sensei_course_progress_find_course_id` - Filter the course ID for a course progress we want to find.
* `sensei_lesson_progress_create_lesson_id` - Filter lesson id for lesson progress creation.
* `sensei_lesson_progress_get_lesson_id` - Filter lesson id for lesson progress creation.
* `sensei_lesson_progress_has_lesson_id` - Filter lesson id for lesson progress check.
* `sensei_lesson_progress_delete_for_lesson_lesson_id` - Filter lesson id for lesson progress deletion.
* `sensei_lesson_progress_count_course_id` - Filter course id for lesson progress counting.
* `sensei_lesson_progress_find_lesson_id` - Filter lesson id when finding lesson progress.
* `sensei_quiz_progress_create_quiz_id` - Filter quiz id for quiz progress creation.
* `sensei_quiz_progress_get_quiz_id` - Filter quiz id for quiz progress retrieval.
* `sensei_quiz_progress_has_quiz_id` - Filter quiz id for quiz progress existence check.
* `sensei_quiz_progress_delete_for_quiz_quiz_id` - Filter quiz id for quiz progress deletion.
* `sensei_quiz_progress_find_quiz_id` - Filter quiz id for quiz progress retrieval.

## Pre-Merge Checklist
